hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/AbstractKeyDeletingService.java

```diff
@@ -388,7 +393,7 @@ protected PurgePathRequest prepareDeleteDirRequest(
     // limit. If count reached limit then there can be some more child
     // paths to be visited and will keep the parent deleted directory
     // for one more pass.
-    String purgeDeletedDir = remainNum > 0 ? delDirName : null;
+    String purgeDeletedDir = (remainNum > 0 && remainingBufLimit > 0) ? delDirName :  null;
```

Comment:
acc to current logic if remainBufLimit = 5 it will include the record with recordSize 10 and the remainBufLimit would be updated to 5-10 = -5. With this the above logic will work . We already maintain some extra buffer when initialising remainBufLimit to handle overflow.
anyways I will need to change the logic while incorporating other comments so this would be taken care of. Thanks